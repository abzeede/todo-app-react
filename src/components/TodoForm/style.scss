.todo-form {
  background-color: rgba(83, 97, 127, 0.75);
  padding: 0.5em;
  display: flex;
  align-items: center;
  border-radius: 4px;
  color: #fff;

  &__plus-icon {
    font-size: 1.5em;
    margin: 0 0.3em;
  }

  &__input {
    display: flex;
    flex: 1;
    height: 1.5em;
    font-size: 1em;
    overflow: hidden;
    box-sizing: border-box;
    border: none;
    background-color: transparent;
    padding: 0 0.5em;
    color: #fff;

    &::placeholder {
      color: #b6bcc7;
    }

    &:focus {
      outline: none;
    }
  }
}
