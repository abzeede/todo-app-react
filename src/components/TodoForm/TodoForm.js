import React from 'react'
import { func, string } from 'prop-types'
import './style.scss'

class TodoForm extends React.Component {
  state = {
    value: '',
  }

  handelInputChange = e => {
    this.setState({ value: e.target.value })
  }

  onSubmit = e => {
    e.preventDefault()
    if (this.state.value !== '') {
      this.props.onSubmit(this.state.value)
      this.setState({ value: '' })
    }
  }

  render() {
    return (
      <form className="todo-form" onSubmit={this.onSubmit}>
        <span className="todo-form__plus-icon">+</span>
        <input
          className="todo-form__input"
          onChange={this.handelInputChange}
          value={this.state.value}
          placeholder={this.props.placeholder}
        />
      </form>
    )
  }
}

TodoForm.propTypes = {
  onSubmit: func.isRequired,
  placeholder: string,
}

TodoForm.defaultProps = {
  placeholder: '',
}

export default TodoForm
