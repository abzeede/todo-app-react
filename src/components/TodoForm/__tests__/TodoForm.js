import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import TodoForm from '../TodoForm'

describe('TodoFrom', () => {
  it('should render properly', () => {
    const wrapper = mount(<TodoForm onSubmit={() => {}} placeholder="add a todo" />)
    const input = wrapper.find('input')

    expect(input.prop('placeholder')).toEqual('add a todo')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should return value when create todo', () => {
    const onSubmit = jest.fn(value => value)
    const wrapper = mount(<TodoForm onSubmit={onSubmit} placeholder="test" />)
    const input = wrapper.find('input')
    const form = wrapper.find('form')

    input.simulate('change', { target: { value: 'I have something to do.' } })
    form.simulate('submit')

    expect(onSubmit).toHaveBeenCalledWith('I have something to do.')
    expect(wrapper.state().value).toBe('')
  })
})
