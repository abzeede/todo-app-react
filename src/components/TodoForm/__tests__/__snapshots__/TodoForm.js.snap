// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoFrom should render properly 1`] = `
<TodoForm
  onSubmit={[Function]}
  placeholder="add a todo"
>
  <form
    className="todo-form"
    onSubmit={[Function]}
  >
    <span
      className="todo-form__plus-icon"
    >
      +
    </span>
    <input
      className="todo-form__input"
      onChange={[Function]}
      placeholder="add a todo"
      value=""
    />
  </form>
</TodoForm>
`;
